enclave-tls/samples/enclave-tls-client/client.c

```diff
@@ -38,20 +38,49 @@ static sgx_enclave_id_t load_enclave(bool debug_enclave)
 	int updated = 0;
 	int ret = sgx_create_enclave(ENCLAVE_FILENAME, debug_enclave, &t, &updated, &eid, NULL);
 	if (ret != SGX_SUCCESS) {
-		ETLS_ERR("Failed to load enclave %d\n", ret);
+		printf("Failed to load enclave %d\n", ret);
 		return -1;
 	}
 
-	ETLS_DEBUG("Success to load enclave id %ld\n", eid);
+	printf("Success to load enclave id %ld\n", eid);
 
-	return eid;
+	return (int64_t)eid;
```

Comment:
Not sure why changing all above things.

Reply:
compile warning if no cast

Reply:
I mean why drop enclave id type.

Reply:
In load_enclave, if failed to create enclave, it will return -1, but sgx_enclave_id_t is unint64 type. Some build warning will be generated. Now I will return 0, if failed to create enclave. It will avoid buiding warning issue.
 
static sgx_enclave_id_t load_enclave(bool debug_enclave)
 33 {
 34         sgx_launch_token_t t;
 35
 36         memset(t, 0, sizeof(t));
 37
 38         sgx_enclave_id_t eid;
 39         int updated = 0;
 40         int ret = sgx_create_enclave(ENCLAVE_FILENAME, debug_enclave, &t, &updated, &eid, NULL);
 41         if (ret != SGX_SUCCESS) {
 42                 printf("Failed to load enclave %d\n", ret);
 43                 return 0;
 44         }
 45
 46         printf("Success to load enclave id %ld\n", eid);
 47
 48         return eid;
 49 }

Reply:
Sound good. Use 0 as the indicator of invalid enclave id and remain the use of sgx_enclave_id_t, right?

Reply:
correct